### PR TITLE
feat(core): reworked the walledgarden layout

### LINKED ISSUE #10305: Reworking the walledgarden pages
I would like to rework the walledgarden pages... something like below.

walledgarden

I started a discussion on elgg.org to get users to provide fanart for the background image:
https://elgg.org/discussion/view/2553331/looking-for-full-hd-fanart-of-elgg

Benefits of this rework would be:
 - easier branding by themes
 - better UX on smaller screens
 - reduce the fancy javascript for content replacement

### PR DESCRIPTION
**Pull request by jdalsem**
Most notable changes
- different look and feel
- dropped the javascript form loading behaviour (including the helper views for ajax loaded forms)
- dropped the walledgarden rounded corners images
- reduced the need for custom css in themes related to walled garden
- moved walled garden css loading to page shell (so no need to load it whereever you draw the walled garden)
- simplyfied and standardized usage of walled_garden pageshell/layout/module

Background Image used is CC0 Public Domain https://pixabay.com/en/lamp-jetty-wooden-pier-banister-677698/

Themes can easily replace the background image by adding the view /walled_garden.jpg

walled garden

fixes #10305

### CODE REVIEW

File: views/default/page/walled_garden.php [hunk 1]
```diff
@@ -9,29 +9,45 @@
  * @uses $vars['sysmessages'] A 2d array of various message registers, passed from system_messages()
  */
```
Comment (mrclay) [inline views/default/page/walled_garden.php]: I wonder if we render some other view to reduce duplication here. Maybe not.

File: mod/externalpages/views/default/expages/wrapper.php [hunk 2]
```diff
@@ -5,12 +5,17 @@
...
 
-echo $vars['content'];
+echo elgg_extract('content', $vars);
+
+$referer = elgg_extract('HTTP_REFERER', $_SERVER);
+if (strpos($referer, elgg_get_site_url()) !== 0) {
+	return;
+}
 
 echo '<div class="mtm">';
```
Comment (hypeJunction) [inline mod/externalpages/views/default/expages/wrapper.php]: Make this a clearfix

File: mod/externalpages/start.php [hunk 3]
```diff
@@ -67,19 +67,18 @@ function expages_page_handler($page, $handler) {
...
-	} else {
-		$content .= elgg_echo("expages:notset");
-	}
-	$content = elgg_view('expages/wrapper', array('content' => $content));
+	]);
+	
+	$description = $object ? $object[0]->description : elgg_echo('expages:notset');
+	
+	$content = elgg_view('expages/wrapper', [
+		'content' => elgg_view('output/longtext', ['value' => $description]),
```
Comment (hypeJunction) [inline mod/externalpages/start.php]: Could you break this down into multiple lines, please

File: mod/externalpages/start.php [hunk 4]
```diff
@@ -94,8 +93,7 @@ function expages_page_handler($page, $handler) {
 		$body = elgg_view_layout('one_column', array('title' => $title, 'content' => $content));
```
Comment (hypeJunction) [inline mod/externalpages/start.php]:
```
$layout = 'default';
if (elgg_get_config('walled_garden') && !elgg_is_logged_in()) {
   $layout = 'walled_garden';
}
$body = elgg_view_layout($layout, [
   'title' => $title,
   'content' => $content,
   'sidebar' => false, // or?
]);
```

File: views/default/forms/user/changepassword.php [hunk 5]
```diff
@@ -3,39 +3,27 @@
  * Reset user password form
  */
 
-$text = elgg_autop(elgg_echo('user:changepassword:change_password_confirm'));
+echo elgg_view('output/longtext', ['value' => elgg_echo('user:changepassword:change_password_confirm')]);
```
Comment (hypeJunction) [inline views/default/forms/user/changepassword.php]: Break down into multiple lines

File: views/default/forms/user/changepassword.php [hunk 6]
```diff
@@ -3,39 +3,27 @@
...
-	'name' => 'password1',
-));
-
-$password2_label = elgg_echo('user:password2:label');
-$password2 = elgg_view('input/password', array(
-	'name' => 'password2',
-));
-
-$u = elgg_view('input/hidden', array(
+echo elgg_view_input('hidden', [
```
Comment (hypeJunction) [inline views/default/forms/user/changepassword.php]: Update to `elgg_view_filed()` unless it hasn't been yet

File: views/default/forms/user/changepassword.php [hunk 7]
```diff
@@ -3,39 +3,27 @@
...
-$u $c
-$submit
-</div>
-HTML;
+echo elgg_view_input('password', [
+	'label' => elgg_echo('user:password2'),
+	'name' => 'password2',
+]);
 
+$footer = elgg_view('input/submit', ['value' => elgg_echo('changepassword')]);
```
Comment (hypeJunction) [inline views/default/forms/user/changepassword.php]: `elgg_view_field()`

File: views/default/forms/user/requestnewpassword.php [hunk 8]
```diff
@@ -5,20 +5,16 @@
...
+
+echo elgg_view_input('text', [
+	'label' => elgg_echo('loginusername'),
+	'name' => 'username',
+	'autofocus' => true,
+]);
+
+echo elgg_view('input/captcha');
+
+$footer = elgg_view('input/submit', array('value' => elgg_echo('request')));
```
Comment (hypeJunction) [inline views/default/forms/user/requestnewpassword.php]: `elgg_view_field()`

File: views/default/page/layouts/walled_garden.php [hunk 9]
```diff
@@ -2,15 +2,11 @@
...
-echo elgg_view_module('walledgarden', '', $vars['content'], array(
-	'class' => $class,
-	'id' => elgg_extract('id', $vars, ''),
-	'header' => ' ',
-	'footer' => ' ',
-));
+$title = elgg_extract('title', $vars);
+$body = elgg_extract('content', $vars);
+
+echo elgg_view_module('walled_garden', $title, $body);
```
Comment (hypeJunction) [inline views/default/page/layouts/walled_garden.php]: Is this needed? We are passing `title` to layout view

File: views/default/page/walled_garden.php (hunk 1 above)
Comment (hypeJunction) [inline views/default/page/walled_garden.php]: I think using the default shell should be fine

File: views/default/resources/account/change_password.php [hunk 10]
```diff
@@ -32,7 +32,6 @@
 $content = elgg_view_form('user/changepassword', array('class' => 'elgg-form-account'), $params);
 
 if (elgg_get_config('walled_garden')) {
-	elgg_load_css('elgg.walled_garden');
 	$body = elgg_view_layout('walled_garden', array('content' => $content));
```
Comment (hypeJunction) [inline views/default/resources/account/change_password.php]: I think we should just pass the title into the layout and get rid of walled garden module wrapper.

File: views/default/resources/account/forgotten_password.php [hunk 11]
```diff
@@ -1,29 +1,25 @@
...
- * @subpackage Registration
  */
 
 if (elgg_is_logged_in()) {
 	forward();
 }
 
-$title = elgg_echo("user:password:lost");
+$title = elgg_echo('user:password:lost');
+$content = elgg_view_form('user/requestnewpassword', ['class' => 'elgg-form-account']);
```
Comment (hypeJunction) [inline views/default/resources/account/forgotten_password.php]: Pass the `$vars` to the form

File: views/default/walled_garden.css.php [hunk 12]
```diff
@@ -2,82 +2,65 @@
...
-.elgg-walledgarden-single > .elgg-head {
-	background: url(<?= elgg_get_simplecache_url("walled_garden/one_column_top.png"); ?>) no-repeat left top;
-}
-.elgg-walledgarden-single > .elgg-body {
-	background: url(<?= elgg_get_simplecache_url("walled_garden/one_column_middle.png"); ?>) repeat-y left top;
-}
-.elgg-walledgarden-single > .elgg-foot {
-	background: url(<?= elgg_get_simplecache_url("walled_garden/one_column_bottom.png"); ?>) no-repeat left top;
+body {
+	background-image: url('<?php echo elgg_get_simplecache_url('walled_garden.jpg'); ?>');
```
Comment (hypeJunction) [inline views/default/walled_garden.css.php]: We don't really need to use php for this. 
`background-image: url(walled_garden.jpg);`

Comment (hypeJunction) [inline views/default/walled_garden.css.php]: I want to eventually make these proper .css files - there is not reason for all the hacky `<style>` declarations.

File: views/default/walled_garden.css.php [hunk 13]
```diff
@@ -2,82 +2,65 @@
...
-}
-.elgg-col:first-child > .elgg-inner {
-	margin: 0 5px 0 0;
-}
-.elgg-col > .elgg-inner {
-	padding: 0 8px;
+.elgg-page-walled-garden {
+	position: relative;
+	right: 0;
+	width: 500px;
```
Comment (hypeJunction) [inline views/default/walled_garden.css.php]: Do we need both `float` and `right`?

**Review by hypeJunction — CHANGES_REQUESTED** [review]

File: views/default/forms/user/changepassword.php (hunk 6 above)
Comment (jdalsem) [inline views/default/forms/user/changepassword.php]: was not available in master at that time...

File: views/default/resources/account/change_password.php (hunk 10 above)
Comment (jdalsem) [inline views/default/resources/account/change_password.php]: so passing title to layout would draw a title (just as the regular shell does) and completely drop the walled garden module as it is just normal content? all up for it 👍

File: views/default/walled_garden.css.php (hunk 12 above)
Comment (jdalsem) [inline views/default/walled_garden.css.php]: this php was the only thing not making them proper css files, so if i change the link, i will also change the file extension

Comment (hypeJunction) [inline views/default/walled_garden.css.php]: 👏

File: views/default/forms/user/changepassword.php (hunk 6 above)
Comment (hypeJunction) [inline views/default/forms/user/changepassword.php]: I am going to merge up now

File: views/default/walled_garden.css.php (hunk 13 above)
Comment (jdalsem) [inline views/default/walled_garden.css.php]: nope

File: views/default/page/walled_garden.php (hunk 1 above)
Comment (jdalsem) [inline views/default/page/walled_garden.php]: don't think so... walled garden shell is a lot different then default shell

File: views/default/forms/user/changepassword.php (hunk 6 above)
Comment (hypeJunction) [inline views/default/forms/user/changepassword.php]: All branches are up to date now.

File: views/default/walled_garden.css [hunk 14]
```diff
@@ -0,0 +1,58 @@
+body {
+	background-image: url('walled_garden.jpg');
```
Comment (mrclay) [inline views/default/walled_garden.css]: This only allows replacement with other JPGs. #8721 Not a blocking issue though.

**Comment by jdalsem** [comment]
Addressed all comments; most notable changes:
- using default layout, no more walled garden layout, only walledgarden shell
- no more walled garden modules
- using elgg_view_field for forms

as far as i can see this is ready for a merge...

**Comment by mrclay** [comment]
On register page, short viewports push the .elgg-page-footer menu into the register button.

**Pull request merged**